actors/market/src/lib.rs

```diff
@@ -1065,25 +977,29 @@ where
     RT: Runtime<BS>,
 {
     let mut pieces = Vec::with_capacity(deal_ids.len());
+
+    let st: State = rt.state()?;
+
     for deal_id in deal_ids {
-        let deal = proposals
-            .get(*deal_id)
+        let deal = st
+            .deal_proposals_get(rt.store(), *deal_id)
             .map_err(|e| {
                 e.downcast_default(
                     ExitCode::USR_ILLEGAL_STATE,
                     format!("failed to get deal_id ({})", deal_id),
                 )
             })?
-            .ok_or_else(|| actor_error!(not_found, "proposal doesn't exist ({})", deal_id))?;
+            .ok_or_else(|| actor_error!(not_found, "proposal doesn't exist {}", deal_id))?;
+
         pieces.push(PieceInfo { cid: deal.piece_cid, size: deal.piece_size });
     }
     rt.compute_unsealed_sector_cid(sector_type, &pieces).map_err(|e| {
         e.downcast_default(ExitCode::USR_ILLEGAL_ARGUMENT, "failed to compute unsealed sector CID")
     })
 }
 
-pub fn validate_and_return_deal_space<BS>(
-    proposals: &DealArray<BS>,
+pub fn validate_and_return_deal_space<BS, RT>(
+    rt: &RT,
```

Comment:
Please revert this to take the DealArray parameter. As a general rule, we avoid passing the Runtime down to helper functions where possible, since it's a big dependency that can make testing lower levels hard.

Reply:
Sure, Done 👍🏾